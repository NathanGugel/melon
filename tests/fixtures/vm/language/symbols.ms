let mySymbol = $$;
let mySymbolLabel = $"my symbol"$;

io.print(mySymbol);
io.print(mySymbolLabel);
io.print(mySymbol == mySymbol);
io.print(mySymbolLabel == mySymbolLabel);
io.print(mySymbol == mySymbolLabel);
io.print($$ == $$);
io.print($"same name"$ == $"same name"$);

let anotherVar = mySymbol;

io.print(anotherVar == mySymbol);

let myKeyA = $"the key"$;
let myKeyB = $"the key"$;

let myObj = {};

myObj[myKeyA] = "the value of key a";
myObj[myKeyB] = "the value of key b";

io.print(myObj[myKeyA]);
io.print(myObj[myKeyB]);

let myObj2 = {
    [$$] = "lost forever",
    [myKeyA] = "another value of key a"
};

io.print(myObj2[$$]);
io.print(myObj2[myKeyA]);