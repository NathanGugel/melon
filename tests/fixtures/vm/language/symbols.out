[Symbol]
[Symbol "my symbol"]
true
true
false
false
false
true
the value of key a
the value of key b
null
another value of key a
